homeassistant/components/airzone_cloud/climate.py

```diff
@@ -290,9 +314,58 @@ def __init__(
         ]
         if HVACMode.OFF not in self._attr_hvac_modes:
             self._attr_hvac_modes += [HVACMode.OFF]
+        if (
+            self.get_airzone_value(AZD_SPEED) is not None
+            and self.get_airzone_value(AZD_SPEEDS) is not None
+        ):
+            self._initialize_fan_speeds()
 
         self._async_update_attrs()
 
+    def _initialize_fan_speeds(self) -> None:
+        """Initialize Aidoo fan speeds."""
+        azd_speeds: dict[int, int] = self.get_airzone_value(AZD_SPEEDS)
+        max_speed = max(azd_speeds)
+
+        fan_speeds: dict[int, str]
+        if speeds_map := FAN_SPEED_MAPS.get(max_speed):
+            fan_speeds = speeds_map
+        else:
+            fan_speeds = {}
+
+            for speed in azd_speeds:
+                if speed == 0:
+                    fan_speeds[speed] = FAN_AUTO
+                else:
+                    fan_speeds[speed] = f"{int(round((speed * 100) / max_speed, 0))}%"
```

Comment:
I suggest styling it like this instead, i.e. with a space before the %-sign:
```suggestion
                    fan_speeds[speed] = f"{int(round((speed * 100) / max_speed, 0))} %"
```

Reply:
I prefer to keep it without a whitespace, since that's how I implemented it in the local integration:
https://github.com/home-assistant/core/blob/0b113b6b714cfa9f3c7421081763dbf8d145b546/homeassistant/components/airzone/climate.py#L169